sklearn/metrics/_classification.py

```diff
@@ -234,6 +236,10 @@ def confusion_matrix(y_true, y_pred, *, labels=None, sample_weight=None,
         conditions or all the population. If None, confusion matrix will not be
         normalized.
 
+    cluster_classes : bool, default=False
+        Sort the classes in order for the similar the classes to end up next to
```

Comment:
"similar" -> "most confused" perhaps?

Please state that it changes the return type.

You might also note that this can be beneficial in the case where there are lots of classes.